pkg/registry/apis/provisioning/resources/folders.go

```diff
@@ -141,6 +142,28 @@ func (fm *FolderManager) EnsureFolderExists(ctx context.Context, folder Folder,
 	})
 
 	if _, err := fm.client.Create(ctx, obj, metav1.CreateOptions{}); err != nil {
+		// there is a potential race here where two syncs can be triggered
+		// if we try to create and there is an error, check if it is from another sync
+		// job for this repo that created it
+		if apierrors.IsAlreadyExists(err) || err.Error() == dashboards.ErrFolderVersionMismatch.Error() {
```

Comment:
very nit: For the second check, is `errors.Is(err, dashboards.ErrFolderVersionMismatch)` not working?

Reply:
It isn't :( it is wrapped by the k8s error message